### ISSUE TITLE
SEM-EDS Quantification

### ISSUE THREAD

**Issue by dnjohnstone** [hyperspy/hyperspy#1402]
An obvious absence in HyperSpy is SEM-EDS quantification.

Something should be done about that - @sem-geologist you seem to be doing a lot of SEM-EDS, what are you doing about quantification?

**Comment by sem-geologist** [comment]
To be honest... completely nothing I use and manipulate/calculate everything on "counts" level.
Do I have any plans? yes I do, I want to construct whole PhiRho(Z) thing, as I also need that on WDS on EMPA. I simply have not much time, my PhD deadline is near. I think standardless analysis would be extremely hard to do, even NIST DTSA II provide only standard based analyses...

**Comment by jat255** [comment]
If anything, I would say that standards-based quantification should be implemented rather than standardless. There are a number of studies (D. Newbury and others) showing how error-prone commercial standardless systems are done. Also, making inaccurate standardless quantification more widely accessible will only exacerbate the existing issue of poor EDS data presented in the literature (personal opinion).

**Comment by sem-geologist** [comment]
The problem with standard EDS is that many many many many.... microscopes have no means to measure and/or track the beam stability during analysis, which using o standards makes completely meaningless (i.e. field emission guns are so insanely unstable). Actually I am not in such bad opinion about standardless analysis (Bruker ones are quite satisfactory), in geological context it is satisfactory in many cases to distinguish minerals, of course it would completely be no-go for geothermobarometry or geochronology or similar stuff where it is used for some calculations.

**Comment by sem-geologist** [comment]
I just came across this:
http://pypenelope.sourceforge.net/features.html
looks interesting

**Comment by sem-geologist** [comment]
I just am not sure if they have some binary blobs :/

**Comment by sem-geologist** [comment]
yes, the PENELOPE itself is not open source... :/

**Comment by jeinsle** [comment]
While Standards based EDS is the most accurate approach to SEM-EDS quantification, there exist many questions where being able to get a rough estimate tells you a) if you have something interesting  and b) then sets you up for further analysis using a higher accuracy technique like WDS, or standards EDS.  

In the case that this came up with in my discussions with Duncan,  we thought a simple 'Casting approximation' would be useful for first order estimates, especially in cases like mine where I am usually collecting additional analytical data about a sample like EBSD for mineral identification.   What I would be hoping for from this discussion, is if there is someone that could do the development, as well I am not really a code developer, but a great code tester.  I currently have about four to ten different data sets that could use some low-level quantification so we can progress. Hence, I have a lot of data to test the quantification again, I just need some help getting it implemented into the Hyperspy library.

thanks

**Comment by sem-geologist** [comment]
Ok, I think I could try doing that, however we need some info:

* EDS window specifications (SLEW3.3 SLEW 3.7, beryllium and other type of the most common windows) I guess this info could be extractable from DTSA II (is it ethical? they only have note that they are in public domain, how far are they Open Source?)
* Detector effectiveness vs Energy? (have no idea where to get those)

Those are one of the most influential to net counts.

**Comment by jeinsle** [comment]
Hiya Petras-

That sounds great.

Francisco has suggested that we should talk about a project that I am 
working on that he thinks your work would be a nice contribution to.  If 
you could drop me an email maybe we could set up a skype call?

Cheers

Josh

**Comment by sem-geologist** [comment]
@jeinsle , I see nowhere your email and I can't get it from git as I see no commits, however You can extract my email from commits history...

**Comment by jeinsle** [comment]
ahh right,

mine is jfe26@cam.ac.uk <mailto:jfe26@cam.ac.uk>

that said I will fire one off to your university address now if that is cool?


Joshua F. Einsle
Post-doctoral Research Associate
Nanopaleomagnetism
jfe26@cam.ac.uk
http://www.esc.cam.ac.uk/directory/joshua-einsle <http://www.esc.cam.ac.uk/directory/joshua-einsle>
Department of Earth Sciences
Department of Materials Science and Metallurgy
University of Cambridge

**Comment by sem-geologist** [comment]
recently I found this library:
https://github.com/tschoonj/xraylib

they have BSD license. Would BSD license be the problem?
I still need to look more into that library, but it looks active developed, and it deals with xray absorbtion.

**Comment by francisco-dlp** [comment]
BSD license is not a problem—it's de most permissive around so it mixes well with everything.

**Comment by pburdet** [comment]
You need to contact [ppinard](https://github.com/ppinard) if you like to add  EDX/SEM quantification and simulation in hyperspy. He develops a code to link python and all the main software of EDX (DTSA, Peneloppe,...). He is promoting python and open development in the WDX/EDX community.

**Comment by sem-geologist** [comment]
Yes I know that, but he do the wrapper only, but Peneloppe code itself is not open, it is only binary blobs, that is going to be a pain.
